testrepository is not needed to run tests.
We do not have it packaged.

--- pbr-7.0.3/test-requirements.txt.orig
+++ pbr-7.0.3/test-requirements.txt
@@ -20,4 +20,3 @@
 # optionally exposed by distutils commands
 sphinx!=1.6.6,!=1.6.7,>=1.6.2,<2.0.0;python_version=='2.7' # BSD
 sphinx!=1.6.6,!=1.6.7,>=1.6.2;python_version>='3.4' # BSD
-testrepository>=0.0.18 # Apache-2.0/BSD
